### PR TITLE
Token Logic (Models Update)

### PR DESCRIPTION
**Pull request by tsource98**

### CODE REVIEW

File: lib/mix/tasks/gen.badges.ex [hunk 1]
```diff
@@ -59,7 +59,8 @@ defmodule Mix.Tasks.Gen.Badges do
           description: description,
           begin: begin_datetime,
           end: end_datetime,
-          type: String.to_integer(type)
+          type: String.to_integer(type),
+          tokens: tokens
```
Comment (joaop21) [inline lib/mix/tasks/gen.badges.ex:63]: tokens should be casted to integer

File: lib/safira/contest/badge.ex [hunk 2]
```diff
@@ -36,7 +38,7 @@ defmodule Safira.Contest.Badge do
   @doc false
   def admin_changeset(badge, attrs) do
     badge
-    |> cast(attrs, [:begin, :end, :name, :description, :type])
+    |> cast(attrs, [:begin, :end, :name, :description, :type, :tokens])
     |> cast_attachments(attrs, [:avatar])
     |> validate_required([:begin, :end, :name, :description, :type])
```
Comment (joaop21) [inline lib/safira/contest/badge.ex:43]: should include `:tokens`

File: lib/safira_web/views/attendee_view.ex [hunk 3]
```diff
@@ -23,7 +23,8 @@ defmodule SafiraWeb.AttendeeView do
       avatar: Avatar.url({attendee.avatar, attendee}, :original),
       badges: render_many(attendee.badges, SafiraWeb.BadgeView, "badge.json"),
       badge_count: attendee.badge_count,
-      volunteer: attendee.volunteer
+      volunteer: attendee.volunteer,
+      total_tokens: attendee.total_tokens,
```
Comment (joaop21) [inline lib/safira_web/views/attendee_view.ex:27]: I think token_balance would be more adequate.

File: lib/safira_web/views/attendee_view.ex [hunk 4]
```diff
@@ -46,7 +47,8 @@ defmodule SafiraWeb.AttendeeView do
       nickname: attendee.nickname,
       name: attendee.name,
       avatar: Avatar.url({attendee.avatar, attendee}, :original),
-      volunteer: attendee.volunteer
+      volunteer: attendee.volunteer,
+      total_tokens: attendee.total_tokens,
```
Comment (joaop21) [inline lib/safira_web/views/attendee_view.ex:51]: same as above

**Review by joaop21 — CHANGES_REQUESTED** [review]
You should run `mix format` against the changed files.

File: mix.exs [hunk 5]
```diff
@@ -24,6 +24,7 @@ defmodule Safira.MixProject do
     ]
   end
 
+
   # Specifies which paths to compile per environment.
```
Comment (jpcorreia99) [inline mix.exs:28]: I think this was, probably, a misclick.

**Comment by tsource98** [comment]
Guess your comments are now fixed. Feel free to suggest anything else

**Review by jpcorreia99 — APPROVED** [review]
:shipit:

File: priv/repo/migrations/20210120134552_add_tokens_badge.exs [hunk 6]
```diff
@@ -0,0 +1,10 @@
+defmodule Safira.Repo.Migrations.AddTokensBadge do
+  use Ecto.Migration
+
+  def change do
+
+    alter table(:badges) do
+      add :tokens, :integer
+  end
+end
+end
```
Comment (joaop21) [inline priv/repo/migrations/20210120134552_add_tokens_badge.exs:10]: You should run `mix format priv/repo/migrations/20210120134552_add_tokens_badge.exs`

File: priv/repo/migrations/20210207173118_add_token_balance_attendee.exs [hunk 7]
```diff
@@ -0,0 +1,10 @@
+defmodule Safira.Repo.Migrations.AddTokenBalanceAttendee do
+  use Ecto.Migration
+
+  def change do
+
+    alter table(:attendees) do
+      add :token_balance, :integer
+  end
+end
+end
```
Comment (joaop21) [inline priv/repo/migrations/20210207173118_add_token_balance_attendee.exs:10]: You should run `priv/repo/migrations/20210207173118_add_token_balance_attendee.exs`

**Review by joaop21 — CHANGES_REQUESTED** [review]

**Review by jpcorreia99 — APPROVED** [review]

**Review by joaop21 — APPROVED** [review]

**Pull request merged**